cmd/controller/controller.go

```diff
@@ -190,38 +283,40 @@ func (c *Controller) updateRelease(key string) error {
 
 	helmObj := obj.(*helmCrdV1.HelmRelease)
 
-	// FIXME: make configurable
-	keyring := "/keyring/pubring.gpg"
-
-	dl := downloader.ChartDownloader{
-		HelmHome: settings.Home,
-		Out:      os.Stdout,
-		Keyring:  keyring,
-		Getters:  getter.All(settings),
-		Verify:   downloader.VerifyNever, // FIXME
-	}
-
 	repoURL := helmObj.Spec.RepoURL
 	if repoURL == "" {
 		// FIXME: Make configurable
 		repoURL = defaultRepoURL
 	}
 
-	certFile := ""
-	keyFile := ""
-	caFile := ""
-	chartURL, err := repo.FindChartInRepoURL(repoURL, helmObj.Spec.ChartName, helmObj.Spec.Version, certFile, keyFile, caFile, getter.All(settings))
+	authHeader := ""
+	if helmObj.Spec.Auth.Header != nil {
+		namespace := os.Getenv("POD_NAMESPACE")
+		if namespace == "" {
+			namespace = defaultNamespace
+		}
+		secret, err := c.kubeClient.Core().Secrets(namespace).Get(helmObj.Spec.Auth.Header.SecretKeyRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		var b bytes.Buffer
+		b.Write(secret.Data[helmObj.Spec.Auth.Header.SecretKeyRef.Key])
+		authHeader = b.String()
+	}
+
+	log.Printf("Downloading repo %s index...", repoURL)
+	repoIndex, err := fetchRepoIndex(repoURL, authHeader)
 	if err != nil {
 		return err
 	}
 
-	log.Printf("Downloading %s ...", chartURL)
-	fname, _, err := dl.DownloadTo(chartURL, helmObj.Spec.Version, settings.Home.Archive())
+	chartURL, err := findChartInRepoIndex(repoIndex, helmObj.Spec.ChartName, helmObj.Spec.Version)
```

Comment:
Afaik, if `chartURL` is relative, it needs to be resolved relative to `repoURL`.  Something like:
```go
func resolveURL(base, ref string) (string, error) {
    baseURL, err := url.Parse(base)
    if err != nil {
        return "", err
    }
    refURL, err := base.Parse(ref)
    if err != nil {
        return "", err
    }
    return refURL.String(), nil
}

...
chartURL, err = resolveURL(repoURL, chartURL)
if err != nil {
    return err
}
```

Reply:
fixed! thanks for picking up on that.